src/misc/fs_utils_posix.cpp

```diff
@@ -354,4 +357,104 @@ uint16_t local_drive_set_attributes(const std_fs::path& path,
 	return status ? DOSERR_NONE : DOSERR_ACCESS_DENIED;
 }
 
+// POSIX does not guarantee to read or write all bytes requested at once.
+// On Linux, it will read the entire chunk in one call (assuming it's a normal file).
+// On Windows, it does not always do so (although we're not using these functions on Windows).
+// On Mac and the various BSDs, who knows. So be safe and do a loop.
```

Comment:
Nice explanation 👍🏻 And yeah, this is the correct way to read a file on all platforms. Even Linux might change things up tomorrow.

Reply:
Linux is pretty strict about not breaking userspace so I doubt this behavior will change.  I suppose someone could mount their DOSBox drive on a networked filesystem or something though and then I'm not sure what the behavior is but the loop should work everywhere.